internal/dag/cache.go

```diff
@@ -103,19 +103,9 @@ func (kc *KubernetesCache) Insert(obj interface{}) bool {
 	maybeInsert := func(obj interface{}) bool {
 		switch obj := obj.(type) {
 		case *v1.Secret:
-			valid, err := isValidSecret(obj)
-			if !valid {
-				if err != nil {
-					kc.WithField("name", obj.GetName()).
-						WithField("namespace", obj.GetNamespace()).
-						WithField("kind", "Secret").
-						WithField("version", k8s.VersionOf(obj)).
-						Error(err)
-				}
-				return false
-			}
-
-			kc.secrets[k8s.NamespacedNameOf(obj)] = obj
+			// Secret validation status is intentionally cleared, it needs
+			// to be re-validated after an insert.
+			kc.secrets[k8s.NamespacedNameOf(obj)] = &Secret{Object: obj}
```

Comment:
we end up caching more Secrets this way, but we never look at them unless they're actually referenced. I'm not sure it matters all that much memory-wise, since these are all in the controller-runtime cache already anyway, and we're just storing pointers to them.

Reply:
yeah even if it stores a few more references, we do gain some information since you have the validation status cached which stops redoing some work if a secret is used in multiple resources

Reply:
@skriss @sunjayBhatia 

Noticing a regression in Contour's resource usage after upgrading to contour-v1.24.1 (from contour-v1.22.x) . IMO, above change could be the reason ? Contour's memory usage in clusters having very large secret resources (like secrets of type `helm.sh/release.v1` )  keeps growing and leading to OOM due to reaching container's resources.memory.limits.  contour-v1.22.x on the same cluster (with similar K8s resources profile and large secrets) didn't had this issue.

1.) heap pprof from a Contour-v1.24.1 instance shows 3GB+ in use memory towards v1.Secrets resources unmarshalling ( the above change is holding back references to secrets in informer cache, preventing the unmarshall'ed bytes from GC'ed ?) :
![contour-heap](https://user-images.githubusercontent.com/3928411/223648415-d5980938-8c02-4327-a26b-d51c3f78a1c0.png)

2.) memory usage graph (before 03/06 12 pm and after):

<img width="1613" alt="Screen Shot 2023-03-07 at 11 24 14 PM" src="https://user-images.githubusercontent.com/3928411/223649015-3b13d7c9-979c-4f14-b7a1-dcb08a9255f8.png">

3.) Also, the cpu usage of Contour has gone up considerably since the v1.22.x to v1.24.1 upgrade (before 03/06 12pm and after) :

<img width="1606" alt="Screen Shot 2023-03-07 at 11 24 26 PM" src="https://user-images.githubusercontent.com/3928411/223649180-ce639533-f30b-4ed8-b05a-2dfe8ffe147d.png">


Also cc @tsaarni (who's related work on filtering out unwanted Secrets from the informer cache is also relevant, IMO).

Reply:
@ravilr sounds like a definite possibility. One change we could make here is to re-introduce a filter to only cache Secrets of type Opaque and TLS. I will also take a further look at @tsaarni's work here.

Reply:
On further debugging, this turned out to be a [bug in a controller causing it to patch Secret resources in a tight loop(more than 20 times per second)](https://github.com/cert-manager/cert-manager/issues/5755).  since the cluster in question went through the upgrade of multiple components including Contour,  the change in contour's resource usage was wrongly attributed to contour upgrade change. 

but, IMO, contour should still filter out unrelated secrets like helm release secrets which are very large, from its secret  informer cache, to prevent contour being affected with high resource usage and event processing overhead from those unrelated Secrets resources in a cluster. So, I hope we can take [Tero's PR ](https://github.com/projectcontour/contour/pull/5099)forward.